python/hsfs/feature_view.py

```diff
@@ -3632,22 +3643,23 @@ def log(
             self.enable_logging()
         return self._feature_view_engine.log_features(
             self,
+            self.feature_logging,
             features,
             predictions,
             transformed,
             write_options,
             training_dataset_version=(
                 training_dataset_version or self.get_last_accessed_training_dataset()
             ),
-            hsml_model=hsml_model,
```

Comment:
Another improvement can be to check if `training_dataset_version or self.get_last_accessed_training_dataset()` is still None, and if a model is provided, we get the training dataset version using provenance.
Then the training dataset version is optional if the model is passed, so simpler and less arguments.